### PR TITLE
add(unit-test): Write unit test of all components

### PR DESCRIPTION
**Pull request by syedsaadqamar**
#25

### CODE REVIEW

**Comment by avatsaev** [comment]
Thanks, reviewing

File: src/app/views/contacts/contact-details/contact-details.component.ts [hunk 1]
```diff
@@ -19,8 +19,8 @@ export class ContactDetailsComponent implements OnInit, OnDestroy {
 
   constructor(
       private activatedRoute: ActivatedRoute,
-      private router: Router,
-      private contactsFacade: ContactsStoreFacade,
+      public router: Router,
```
Comment (avatsaev) [inline src/app/views/contacts/contact-details/contact-details.component.ts]: Keep fields private, use dependency resolvers to get injected values: 
https://codecraft.tv/courses/angular/unit-testing/dependency-injection/#_resolving_via_the_component_injector

File: src/app/views/contacts/contact-edit/contact-edit.component.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ export class ContactEditComponent implements OnInit, OnDestroy {
   constructor(
     private activatedRoute: ActivatedRoute,
     private router: Router,
-    private contactsFacade: ContactsStoreFacade,
+    public contactsFacade: ContactsStoreFacade,
```
Comment (avatsaev) [inline src/app/views/contacts/contact-edit/contact-edit.component.ts]: Keep fields private, use dependency resolvers to get injected values: 
https://codecraft.tv/courses/angular/unit-testing/dependency-injection/#_resolving_via_the_component_injector

File: src/app/views/contacts/contact-new/contact-new.component.ts [hunk 3]
```diff
@@ -12,8 +12,8 @@ import { ContactsStoreFacade} from '@app/contacts-store/contacts-store.facade';
 export class ContactNewComponent implements OnInit, OnDestroy {
 
   constructor(
-    private contactsFacade: ContactsStoreFacade,
-    private router: Router
+    public contactsFacade: ContactsStoreFacade,
```
Comment (avatsaev) [inline src/app/views/contacts/contact-new/contact-new.component.ts]: Keep fields private, use dependency resolvers to get injected values: 
https://codecraft.tv/courses/angular/unit-testing/dependency-injection/#_resolving_via_the_component_injector

File: src/app/views/contacts/contacts-index/contacts-index.component.ts [hunk 4]
```diff
@@ -14,7 +14,7 @@ export class ContactsIndexComponent implements OnInit {
 
   contacts$ = this.contactsFacade.contacts$;
 
-  constructor(public contactsFacade: ContactsStoreFacade, private router: Router) { }
+  constructor(public contactsFacade: ContactsStoreFacade, public router: Router) { }
```
Comment (avatsaev) [inline src/app/views/contacts/contacts-index/contacts-index.component.ts]: Keep fields private, use dependency resolvers to get injected values: 
https://codecraft.tv/courses/angular/unit-testing/dependency-injection/#_resolving_via_the_component_injector

File: src/app/views/contacts/contact-details/contact-details.component.ts (hunk 1 above)
Comment (syedsaadqamar) [reply src/app/views/contacts/contact-details/contact-details.component.ts]: @avatsaev thanks for sharing a helpful knowledge, now I am updating the PR.

File: src/app/views/contacts/contact-details/contact-details.component.spec.ts [hunk 5]
```diff
@@ -15,11 +15,14 @@ import {ContactsService} from '@app/core/services/contacts.service';
 import {HttpClientTestingModule} from '@angular/common/http/testing';
```
Comment (avatsaev) [inline src/app/views/contacts/contact-details/contact-details.component.spec.ts:15]: unused import

**Review by avatsaev — APPROVED** [review]

**Comment by avatsaev** [comment]
very well done thanks

**Pull request merged**